addons/toolbars/README.md

```diff
@@ -63,14 +72,25 @@ You should see a dropdown in your toolbar with options `light` and `dark`.
 
 Now, let's wire it up! We can consume our new `theme` global arg in a decorator using the `context.globalArgs.theme` value.
 
-For example, suppose you are using`styled-components`. You can add a theme provider decorator to your `.storybook/preview.js` config:
+For example, suppose you are using `styled-components`. You can add a theme provider decorator to your `.storybook/preview.js` config:
 
-```js
-const styledComponentsThemeDecorator = (storyFn, { globalArgs: { theme } }) => (
-  <ThemeProvider {...getTheme(theme)}>{storyFn()}</ThemeProvider>
-);
+```ts
+import { ThemeProvider } from 'styled-components';
+import { StoryContext, StoryGetter, StoryWrapper } from '@storybook/addons';
+
+const withThemeProvider: StoryWrapper = (Story: StoryGetter, context: StoryContext) => {
+  // context.globalArgs.theme here will be either 'light' or 'dark'
+  // getTheme being a function retrieving the actual theme object from that value
+  const theme = getTheme(context.globalArgs.theme);
+
+  return (
+    <ThemeProvider theme={theme}>
+      <Story {...context} />
```

Comment:
Leaving that spread for now since it seems to be needed with TypeScript (see #10579)

Reply:
Ok, I'm not happy with the typings here but I think that's a separate issue. cc @tmeasday @wKich

Reply:
I would be interested to know how this can be avoided as well then :)

Reply:
Actually, it seems that `StoryWrapper` as a return type does not work in the end.
 
Following the types warnings, I only come with `DecoratorFunction<any>`, would really love some help on that 😅

Reply:
I think, we could try to solve typings issue as part of the integration common types https://github.com/storybookjs/csf